### PR TITLE
Adds guards for possibly undefined properties

### PR DESCRIPTION
**Pull request by alanbo**
*Issue #, if available:*
The function validating parameters was missing type guards.

*Description of changes:*
I've added missing type guards.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/validation/validateStates.ts [hunk 1]
```diff
@@ -62,7 +62,7 @@ function validateParameters(parametersPropNode: PropertyASTNode, document: TextD
 
                     diagnostics.push(Diagnostic.create(range, MESSAGES.INVALID_JSON_PATH, DiagnosticSeverity.Error))
                 }
-            } else if (isObjectNode(prop.valueNode)) {
+            } else if (prop.valueNode && isObjectNode(prop.valueNode)) {
```
Comment (jormello) [inline src/validation/validateStates.ts:65]: Same comment as above.

File: src/validation/validateStates.ts [hunk 2]
```diff
@@ -51,9 +51,9 @@ function validateParameters(parametersPropNode: PropertyASTNode, document: TextD
     let diagnostics: Diagnostic[] = []
     const valueNode = parametersPropNode.valueNode
 
-    if (isObjectNode(valueNode)) {
+    if (valueNode && isObjectNode(valueNode)) {
```
[the remark this answers is not in the document]
Comment (alanbo) [reply src/validation/validateStates.ts:54]: `valueNode` is possibly undefined

File: src/validation/validateStates.ts [hunk 3]
```diff
@@ -51,9 +51,9 @@ function validateParameters(parametersPropNode: PropertyASTNode, document: TextD
...
         valueNode.properties.forEach(prop => {
-            if (prop.keyNode.value.endsWith('.$')) {
+            if (prop.valueNode && prop.keyNode.value.endsWith('.$')) {
```
[the remark this answers is not in the document]
Comment (alanbo) [reply src/validation/validateStates.ts:56]: no, we are checking different properties of `prop`. One is `valueNode` and the other is `keyNode`.

File: src/validation/validateStates.ts (hunk 1 above)
Comment (alanbo) [reply src/validation/validateStates.ts:65]: `valueNode` is possibly undefined

File: src/validation/validateStates.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jormello) [reply src/validation/validateStates.ts:54]: It seems a little bit weird that isObjectNode( ) still returns true when given an undefined valueNode.

File: src/validation/validateStates.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (jormello) [reply src/validation/validateStates.ts:56]: I was thinking something like 

if (prop.valueNode && prop?.keyNode?.value?.endsWith('.$'))

File: src/validation/validateStates.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (alanbo) [reply src/validation/validateStates.ts:54]: no, it doesn't return true, it throws an error. That's why I added a check before it, it should never get undefined.

File: src/validation/validateStates.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (alanbo) [reply src/validation/validateStates.ts:56]: but that's completely unnecessary, only value node can be undefined. All the properties within `prop.keyNode.value.` are always defined.

**Pull request merged**